cmd/routed-eni-cni-plugin/cni.go

```diff
@@ -283,6 +306,40 @@ func del(args *skel.CmdArgs, cniTypes typeswrapper.CNITYPES, grpcClient grpcwrap
 		return errors.Wrap(err, "del cmd: failed to load k8s config from args")
 	}
 
+	if args.Netns == "" {
+		log.Info("Netns() is empty, so network already cleanedup. Nothing to do")
+		return nil
+	}
+	prevResult, ok := conf.PrevResult.(*current.Result)
+
+	// Try to use prevResult if available
+	// prevResult might not be availabe, if we are still using older cni spec < 0.4.0.
+	// So we should fallback to the old clean up method
+	if ok {
+		dummyVlanInterfaceName := generateHostVethName(dummyVlanInterfacePrefix, string(k8sArgs.K8S_POD_NAMESPACE), string(k8sArgs.K8S_POD_NAME))
+		for _, iface := range prevResult.Interfaces {
+			if iface.Name == dummyVlanInterfaceName {
+				podVlanId, err := strconv.Atoi(iface.Mac)
+				if err != nil {
+					return errors.Wrap(err, "Failed to parse vlanId from prevResult")
+				}
+				// podVlanId == 0 means pod is not using branch ENI
+				// then fallback to existing cleanup
+				if podVlanId == 0 {
```

Comment:
Will we have this situation? I mean dummyVlanInterfaceName is added if SGPP is enabled so we should typically have podVlanID right? if it is 0 then it is an error.

Reply:
Updated

Reply:
If we have an unsupported instance and PPSG is enabled then we can actually have podvlanID as 0. Maybe we need to special case this? @achevuru any thoughts?

Reply:
I see that we only add the `dummyVlanInterface` if `podVlanId != 0` during the `add` flow. So, I think we might be Ok here in the del flow as this is executed only if we find `dummyVlanInterface` in prevResult...